src/libtorch.cc

```diff
@@ -552,11 +582,48 @@ ModelInstanceState::Create(
 
 ModelInstanceState::ModelInstanceState(
     ModelState* model_state, TRITONBACKEND_ModelInstance* triton_model_instance)
+#ifdef TRITON_ENABLE_GPU
+    : BackendModelInstance(model_state, triton_model_instance),
+      model_state_(model_state), device_(torch::kCPU), is_dict_input_(false),
+      pytorch_stream_(at::cuda::getDefaultCUDAStream())
+#else
     : BackendModelInstance(model_state, triton_model_instance),
       model_state_(model_state), device_(torch::kCPU), is_dict_input_(false)
+#endif
 {
   if (Kind() == TRITONSERVER_INSTANCEGROUPKIND_GPU) {
     device_ = torch::Device(torch::kCUDA, DeviceId());
+#ifdef TRITON_ENABLE_GPU
+    pytorch_stream_ =
+        at::cuda::getStreamFromPool(false /* is_high_priority */, DeviceId());
+    if (stream_ != nullptr) {
+      cudaError_t err = cudaStreamDestroy(stream_);
+      if (err != cudaSuccess) {
+        TRITONSERVER_LogMessage(
+            TRITONSERVER_LOG_ERROR, __FILE__, __LINE__,
+            (std::string("~BackendModelInstance: ") + name_ +
+             " failed to destroy cuda stream: " + cudaGetErrorString(err))
+                .c_str());
+      }
+      stream_ = pytorch_stream_.stream();
```

Comment:
I think we must not set  stream_ here like this. Triton would assume the ownership of stream here.

Reply:
The ownership is handled correctly. We set the stream_ to null before the instance is destructed.